monitoring/users-and-sessions.mdx

```diff
@@ -48,137 +54,139 @@ For the Use cases of a chat bot, a Session may be the entire conversation. For a
+If you're building a system where user interactions happen in a sequence (e.g. a chat bot) you'll find it useful to group these interactions into a single session. The (`with_session`) context manager in Baserun allows you to do just that. By providing a user identifier, you can create a new session. By providing a session ID you can continue an existing session. This allows you to group all related traces under either the User or Session, making it easier to debug or analyze the user's journey through your application.
 
 ### Create a new session
 
 <CodeGroup>
 ```python python
+from hashlib import sha256
+
 import baserun
    
-with baserun.with_session(identifier=username_or_id) as session:
+with baserun.with_session(user_identifier=sha256(b"user@example.com").hexdigest(),) as session:
+    # Add business logic including traces in here
+    ...
     # Store session ID to be continued later
     session_id = session.id
     run_chatbot()
 ```
 
 ```typescript typescript
-import { baserun } from 'baserun'
-import OpenAI from 'openai'
-
-const openai = new OpenAI()
+import { baserun } from "baserun";
+import { createHash } from "crypto";
 
 async function main() {
-  await baserun.init()
-  const { data, sessionId } = await baserun.session({
-    async session() {
-      // add business logic including traces in here
-      return openai.chat.completions.create({
-        messages: [
-          {
-            role: 'user',
-            content: `Say "this is a test"`,
-          },
-        ],
-        model: 'gpt-3.5-turbo',
-      })
+  await baserun.init();
+  const { data, sessionId } = await baserun.session(
+    async function () {
+      // Add business logic including traces in here
     },
-    user: 'user@example.com',
-  })
+    {
+      user: createHash("sha256").update("user@example.com").digest("hex"),
+    },
+  );
 
-  console.log({ data, sessionId })
+  // Store session ID to be continued later
+  console.log({ data, sessionId });
 }
 
-main()
+main();
+
 ```
 
 </CodeGroup>
 
 ### Continuing an existing session
 
-Using the Session ID that was collected when the session was started you can then resume that session by providing the session ID to the `with_session` context manager.
+Using the Session ID that was collected when the session was started you can then resume that session by providing the session ID.
 
 ### Full example
 
 <CodeGroup>
+{/* TODO: python code does more complex things than ts code. should it be simplified to match? */}
```

Comment:
Yes, let's make sure both code examples do the same thing, the examples should be minimal examples of the features described above, nothing else.